core/shared/deutschebahn/helper/timetable.py

```diff
@@ -0,0 +1,113 @@
+
+
+import json
+import requests
+
+from bs4 import BeautifulSoup
+from .apiAuthentication import ApiAuthentication
+
+
+class TimetableRequest(): 
+    def __init__(self, station_id, date, hour):
+        self.api = ApiAuthentication("", "")
+        self.url = f"https://apis.deutschebahn.com/db-api-marketplace/apis/timetables/v1/plan/{station_id}/{date}/{hour}"
+
+    def execute(self):
+        response = requests.get(self.url, headers=self.api.get_headers())
+        if response.status_code == 200:
+            soup = BeautifulSoup(response.text, "xml")
+
+            station_element = soup.find('timetable')
+            station_name = station_element.get('station')
+
+            timetable = []
+
+            for entry in soup.find_all('s'):
+                tl = entry.tl.attrs
+                ar = entry.ar.attrs
+                dp = entry.dp.attrs
+
+                timetable.append({
+                    'tl': tl,
+                    'ar': ar,
+                    'dp': dp
+                })
+
+            json_data = {
+                'station': station_name,
+                'timetable': timetable
+            }
+
+            return json_data
+        else:
+            return None
+
+# Component interface
+class Timetable:
```

Comment:
Kannst du Klassen immer in eigene Dateien lagern? Das machst es leichter und übersichtlicher wenn man etwas erweitern möchte. Also z.B. Ordner Models --> /Models/TimeTable.py